tile_generator/package_flags.py

```diff
@@ -194,12 +194,15 @@ def _apply(self, config_obj, package, release):
         # TODO: Remove the dependency on this in templates
         package['is_app'] = True
 
-        for link_type, link in package.get('consumes', {}).iteritems():
+        for link_name, link in package.get('consumes', {}).iteritems():
             release['consumes'] = release.get('consumes', {})
-            release['consumes'][link_type] = link
-            if 'deployment' in link:
-                release['consumes_cross_deployment'] = release.get('consumes_cross_deployment', {})
-                release['consumes_cross_deployment'][link_type] = link
+            release['consumes'][link_name] = link
+            release['consumes_for_deployment'] = release.get('consumes_for_deployment', {})
+            release['consumes_for_deployment'][link_name] = link.copy()
```

Comment:
Is there a reason you are setting this to a copy of the dict link rather than just pointing it to dict?

Reply:
Since there is no method to jumpstart or specify deploy-all/delete-all job spec, or guide the generation of their templates, the tile.yml consumes dict. fields 'type' and 'optional' are used as parameters to assist in generating the correct job spec file and guide template handling for deploy-all and delete-all jobs. So these are "tile-generator" / generate-time needed fields only.

A bosh deployment manifest does not need these field and they may end up posing a risk as an illegal bosh manifest.

Notice the naming of "consumes_for_deployment", it is a copy of the link which I adjust to remove 'type' and 'optional', while no other filtering is done, the only effective fields that guides bosh in resolving a link are its name , 'from' and 'deployment', the type and optional are used by bosh but obtained from the spec file. At least that is what I saw from the documentation and testing I did.